### DISCUSSION TITLE
How to create a tree-shaked bundle of Tabulator?

### DISCUSSION THREAD

**Discussion by ilmari-lauhakangas** [parcel-bundler/parcel#8708] [Q&A] [answered]
I would like to decrease the file size of [Tabulator](https://github.com/olifolkerd/tabulator) by creating bundles that only use the features I need for specific tables. I followed the advice in the docs for [Core With Optional Modules](https://tabulator.info/docs/5.4/install#setup-basic), but I always end up with a bundle containing all the optional modules.

I tried by downloading dist/js/tabulator_esm.js and creating a tabulator_parcel.js file with the contents:

```
import {Tabulator, FormatModule, EditModule} from './tabulator_esm';

Tabulator.registerModule([FormatModule, EditModule]);

var table = new Tabulator("#example-table", {
  //table setup options
});
```

Then I ran

`parcel build tabulator_parcel.js --no-source-maps --no-optimize --log-level verbose`

It doesn't print anything about tree-shaking failing. I am using Parcel 2.8.2.

**Comment by mischnic** [comment, marked as answer]
I think it's not possible with how tabulator_esm.js is structured. 
For example this variable will always include all "modules": https://github.com/olifolkerd/tabulator/blob/9c8c72f2815243e167760383e903918323d285bd/dist/js/tabulator_esm.js#L19972-L20009 (you'll have to have to wait several seconds for the page to load and for the browser to jump to the right line)

**Reply by ilmari-lauhakangas** [reply]
Thanks a lot, I should open an issue with Tabulator!
